src/blog/fast-super.md

```diff
@@ -0,0 +1,123 @@
+
+Previously, accessing a super property (like `super.x`) was implemented via a runtime call. Starting from V8 v9.0, we reuse the [inline cache (IC) system](https://mathiasbynens.be/notes/shapes-ics) in non-optimized code and generate the proper optimized code for super property access, without having to jump to the runtime.
+
+As you can see from the graphs below, super property access used to be an order of magnitude slower than normal property access because of the runtime call. Now we're much closer to being on par.
+
+![Compare super property access to regular property access, optimized](/_img/fast-super/super-opt.svg)
+
+![Compare super property access to regular property access, unoptimized](/_img/fast-super/super-no-opt.svg)
+
+Super property access is difficult to benchmark, since it must happen inside a function. We can't benchmark individual property accesses, but only bigger chunks of work. Thus the function call overhead is included in the measurement. The above graphs somewhat underestimate the difference between super property access and normal property access, but they're accurate enough for demonstrating the difference between the old and new super property access.
+
+In the unoptimized (interpreted) mode, super property access will always be slower than normal property access, since we need to do more loads (reading the home object from the context and reading the `__proto__` from the home object). In the optimized code, we already embed the home object as a constant whenever possible. This could be further improved by embedding its `__proto__` as a constant too.
+
+### Prototypal inheritance and `super`
+
+Let's start from the basics - what does super property access even mean?
+
+```javascript
+class A { }
+A.prototype.x = 100;
+
+class B extends A {
+  m() {
+    return super.x;
+  }
+}
+const b = new B();
+b.m();
+```
+
+Now `A` is the super class of `B` and  `b.m()` returns `100` as you'd expect.
+
+![Class inheritance diagram](/_img/fast-super/inheritance-1.svg)
+
+The reality of [JavaScript's prototypal inheritance](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain) is more complicated:
+
+![Prototypal inheritance diagram](/_img/fast-super/inheritance-2.svg)
+
+We need to distinguish carefully between the `__proto__` and `prototype` properties - they don't mean the same thing! To make it more confusing, the object `b.__proto__` is often referred to as "`b`'s prototype".
+
+`b.__proto__` is the object from which `b` inherits properties. `B.prototype` is the object which will be the `__proto__` of objects created with `new B()`, such as `b`.
+
+`b` has `B.prototype` and `A.prototype` in its prototype chain (that is, the chain of `__proto__` properties), so it can access all properties defined in those objects. The method `m` is a property of `B.prototype`, `B.prototype.m`, and this is why `b.m()` works.
```

Comment:
And, correspondingly, to visualise this next paragraph too:

```suggestion
In turn, `B.prototype` itself has its own `__proto__` property that equals to `A.prototype`. Together, this forms what's called a prototype chain:

```
b.__proto__ ->
B.prototype ->
B.prototype.__proto__ ->
A.prototype ->
[...]
```

Through this chain, `b` can access all properties defined in any of those objects. The method `m` is a property of `B.prototype` — `B.prototype.m` — and this is why `b.m()` works.
```